admin/class-yoast-network-admin.php

```diff
@@ -0,0 +1,235 @@
+	 */
+	public function get_site_choices( $include_empty = false ) {
+		$choices = array();
+
+		if ( $include_empty ) {
+			$choices['-'] = __( 'None', 'wordpress-seo' );
+		}
+
+		$sites = get_sites( array( 'deleted' => 0 ) );
+		foreach ( $sites as $site ) {
+			$choices[ $site->blog_id ] = $site->blog_id . ': ' . $site->domain;
+
+			$site_states = $this->get_site_states( $site );
+			if ( ! empty( $site_states ) ) {
+				$choices[ $site->blog_id ] .= ' [' . implode( ', ', $site_states ) . ']';
+			}
+		}
+
+		return $choices;
+	}
+
+	/**
+	 * Gets the states of a site.
+	 *
+	 * @param WP_Site $site Site object.
+	 *
+	 * @return array Array of $state_slug => $state_label pairs.
+	 */
+	public function get_site_states( $site ) {
+		$available_states = array(
+			'public'   => __( 'public', 'wordpress-seo' ),
+			'archived' => __( 'archived', 'wordpress-seo' ),
+			'mature'   => __( 'mature', 'wordpress-seo' ),
+			'spam'     => __( 'spam', 'wordpress-seo' ),
+			'deleted'  => __( 'deleted', 'wordpress-seo' ),
+		);
+
+		$site_states = array();
+		foreach ( $available_states as $state_slug => $state_label ) {
+			if ( $site->$state_slug === '1' ) {
+				$site_states[ $state_slug ] = $state_label;
+			}
+		}
+
+		return $site_states;
+	}
+
+	/**
+	 * Handles a request in which plugin network options should be updated.
+	 *
+	 * This method works similar to how option updates are handled in `wp-admin/options.php` and
+	 * `wp-admin/network/settings.php`.
+	 *
+	 * @return void
+	 */
+	public function handle_update_options_request() {
+		$option_group = filter_input( INPUT_POST, 'network_option_group', FILTER_SANITIZE_STRING );
+
+		check_admin_referer( "$option_group-network-options" );
+
+		$whitelist_options = Yoast_Network_Settings_API::get()->get_whitelist_options( $option_group );
+
+		if ( empty( $whitelist_options ) ) {
+			wp_die( __( 'Sorry, you are not allowed to modify unregistered network settings.', 'wordpress-seo' ), '', 403 );
+		}
+
+		foreach ( $whitelist_options as $option_name ) {
+			$value = null;
+			if ( isset( $_POST[ $option_name ] ) ) {
+				$value = wp_unslash( $_POST[ $option_name ] );
+			}
+
+			WPSEO_Options::update_site_option( $option_name, $value );
+		}
+
+		$settings_errors = get_settings_errors();
+		if ( empty( $settings_errors ) ) {
+			add_settings_error( $option_group, 'settings_updated', __( 'Settings Updated.', 'wordpress-seo' ), 'updated' );
+		}
+
+		$this->persist_settings_errors();
+
+		$sendback = add_query_arg( 'settings-updated', 'true', wp_get_referer() );
+		wp_safe_redirect( $sendback );
+		exit;
+	}
+
+	/**
+	 * Handles a request in which a site's default settings should be restored.
+	 *
+	 * @return void
+	 */
+	public function handle_restore_site_request() {
+		check_admin_referer( 'wpseo-network-restore' );
+
+		$option_group = 'wpseo_ms';
+
+		$site_id = ! empty( $_POST[ $option_group ]['site_id'] ) ? (int) $_POST[ $option_group ]['site_id'] : 0;
+		if ( ! $site_id ) {
```

Comment:
Is there perhaps a way to do an early return here? The extra nesting of the `if`/`else` statement within the following `else` leads to extra levels of nesting which seems unnecessary.